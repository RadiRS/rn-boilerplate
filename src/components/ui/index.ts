export { default as Text } from './text';
export { default as Button } from './button';
