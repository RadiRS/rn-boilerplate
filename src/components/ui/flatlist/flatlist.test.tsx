import React from 'react';
import { ListRenderItem, Text } from 'react-native';
import { render } from '@testing-library/react-native';

import { FlatList } from '@/components/ui';

describe('<Flatlist />', () => {
  it('should render component correctly', () => {
    const data: any[] = [];
    const renderItem = () => null;

    const rendered = render(
      <FlatList data={data} renderItem={renderItem} />,
    ).toJSON();

    expect(rendered).toMatchSnapshot();
  });

  it('should render list of items correctly', () => {
    const length = 5;
    const data: any[] = new Array(length);
    const renderItem: ListRenderItem<any> = ({ index }) => <Text>{index}</Text>;

    const rendered = render(<FlatList data={data} renderItem={renderItem} />);

    const items = rendered.UNSAFE_queryAllByType(Text);

    expect(items).toHaveLength(length);
  });
});
