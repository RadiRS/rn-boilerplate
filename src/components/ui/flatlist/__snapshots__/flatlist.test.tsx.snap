// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Flatlist /> should render component correctly 1`] = `
<RCTScrollView
  data={[]}
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  showsHorizontalScrollIndicator={false}
  showsVerticalScrollIndicator={false}
  stickyHeaderIndices={[]}
  viewabilityConfigCallbackPairs={[]}
>
  <View />
</RCTScrollView>
`;
